openapi.yaml

```diff
@@ -0,0 +1,787 @@
+          required: false
+          schema:
+            type: string
+        - name: "securableId"
+          in: "query"
+          description: Limit to the specified securable
+          required: false
+          schema:
+            type: string
+      responses:
+        "200":
+          description: "Display retrieved Grants"
+          content:
+            "application/json":
+              schema:
+                $ref: "#/components/schemas/GrantsResponse"
+    post:
+      operationId: createGrant
+      tags:
+        - "grants"
+      responses:
+        "200":
+          description: "Display created Grant"
+          content:
+            "application/json":
+              schema:
+                $ref: "#/components/schemas/GrantsResponse"
+        "400":
+          $ref: "#/components/responses/BadInput"
+        "500":
+          $ref: "#/components/responses/ServerError"
+      requestBody:
+        required: true
+        content:
+          'application/json':
+            schema:
+              $ref: "#/components/schemas/GrantsRequest"
+  /grants/{grantId}:
+    parameters:
+      - name: "grantId"
+        in: "path"
+        description: "Id of Grant to return"
+        required: true
+        schema:
+          type: "string"
+    get:
+      operationId: readGrant
+      tags:
+        - "grants"
+      responses:
+        "200":
+          description: "Display Grant"
+          content:
+            "application/json":
+              schema:
+                $ref: "#/components/schemas/GrantResponse"
+        "404":
+          $ref: "#/components/responses/NotFound"
+        "500":
+          $ref: "#/components/responses/ServerError"
+    put:
+      tags:
+        - "grants"
+      operationId: updateGrant
+      requestBody:
+        required: true
+        content:
+          'application/json':
+            schema:
+              $ref: "#/components/schemas/GrantRequest"
+      responses:
+        "200":
+          description: "Display updated Grant"
+          content:
+            "application/json":
+              schema:
+                $ref: "#/components/schemas/GrantResponse"
+        "400":
+          $ref: "#/components/responses/BadInput"
+        "404":
+          $ref: "#/components/responses/NotFound"
+        "500":
+          $ref: "#/components/responses/ServerError"
+
+    delete:
+      operationId: deleteGrant
+      tags:
+        - "grants"
+      responses:
+        "200":
+          description: "Grant deleted"
+        "404":
+          $ref: "#/components/responses/NotFound"
+        "500":
+          $ref: "#/components/responses/ServerError"
+  /securables:
+    post:
+      operationId: createSecurable
+      tags:
+        - "securables"
+      requestBody:
+        required: true
+        content:
+          'application/json':
+            schema:
+              $ref: "#/components/schemas/SecurableRequest"
+      responses:
```

Comment:
Are there any securable types that can't be added by users?  For example, is it valid for users to add "system" securables?  Those can't be deleted and include things like "the access manager".

400 should be a valid response to such a POST request.